web/src/compose_fade.ts

```diff
@@ -1,17 +1,81 @@
 import $ from "jquery";
 import _ from "lodash";
+import assert from "minimalistic-assert";
 
 import * as compose_fade_helper from "./compose_fade_helper";
 import * as compose_state from "./compose_state";
 import * as message_lists from "./message_lists";
+import type {Message} from "./message_store";
 import * as message_viewport from "./message_viewport";
 import * as people from "./people";
 import * as rows from "./rows";
+import type {TopicLink} from "./types";
 import * as util from "./util";
 
+type AllVisibilityPolicies = {
+    INHERIT: 0;
+    MUTED: 1;
+    UNMUTED: 2;
+    FOLLOWED: 3;
+};
+
+type MessageContainer = {
+    background_color: string;
+    date_divider_html: string;
+    edited_alongside_sender: boolean;
+    edited_in_left_col: boolean;
+    edited_status_msg: boolean;
+    include_recipient: boolean;
+    include_sender: boolean;
+    is_hidden: boolean;
+    mention_classname: string | null;
+    message_edited_notices_alongside_sender: boolean;
+    message_edited_notices_for_status_message: boolean;
+    message_edit_notices_in_left_col: boolean;
+    msg: Message;
+    sender_is_bot: boolean;
+    sender_is_guest: boolean;
+    should_add_guest_indicator_for_sender: boolean;
+    small_avatar_url: string;
+    status_message: boolean;
+    stream_url: string;
+    subscribed: boolean;
+    timestr: string;
+    topic_url: string;
+    unsubscribed: boolean;
+    want_date_divider: boolean;
+};
+
+type MessageGroup = {
+    all_visibility_policies: AllVisibilityPolicies;
+    always_visible_topic_edit: boolean;
+    date: string;
+    date_unchanged: boolean;
+    display_recipient: string;
+    invite_only: boolean;
+    is_private?: boolean;
+    is_stream: boolean;
+    is_subscribed: boolean;
+    is_web_public: boolean;
+    match_topic?: string;
+    message_containers: MessageContainer[];
+    message_group_id: string;
+    on_hover_topic_edit: boolean;
+    recipient_bar_color: string;
+    stream_id: number;
+    stream_privacy_icon_color: string;
+    stream_url: string;
+    topic: string;
+    topic_is_resolved: boolean;
+    topic_links: TopicLink[];
+    topic_url: string;
+    user_can_resolve_topic: boolean;
+    visibility_policy: number;
+};
```

Comment:
Surely these types should live somewhere else; maybe `message_list_view.js`? It's probably worth including comments at the top of the appropriate file noting these should be moved.

Reply:
Sure. Added comments for the respective types.